### PR TITLE
Implement SYNONYM in CREATE TABLE

### LINKED ISSUE #125: Support SYNONYMs in CREATE TABLE
`SYNONYM`s in `CREATE TABLE` has not yet been supported.

https://cloud.google.com/spanner/docs/reference/standard-sql/data-definition-language#create_table
```
CREATE TABLE [ IF NOT EXISTS ] table_name ( [
   { column_name data_type [NOT NULL]
     [ { DEFAULT ( expression ) | AS ( expression ) [ STORED ]
       | table_constraint | synonym_definition ]
     [ options_def ]
     [ location_name STRING(MAX) NOT NULL PLACEMENT KEY ]
   [, ... ]
] ) PRIMARY KEY ( [column_name [ { ASC | DESC } ], ...] )
[, INTERLEAVE IN PARENT table_name [ ON DELETE { CASCADE | NO ACTION } ] ]
[, ROW DELETION POLICY ( OLDER_THAN ( timestamp_column, INTERVAL num_days DAY ) ) ]

...

and synonym_definition is:
    [ SYNONYM (synonym [, synonym...])]
```
Related: #115

### PR DESCRIPTION
**Pull request by apstndb**
This PR implements `SYNONYM` in `CREATE TABLE`.

fix #125

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -1642,10 +1643,22 @@ type CreateTable struct {
...
 
+// Synonym is SYNONYM node in CREATE TABLE
+//
+//	SYNONYM ({.Name | sql})
+type Synonym struct {
+	// pos = Synonym
+	// end = Name.end
+	Synonym token.Pos // position of "SYNONYM" pseudo keyword
+	Rparen  token.Pos
+	Name    *Ident
```
Comment (makenowjust) [inline ast/ast.go:1659]:
```suggestion
	// pos = Synonym
	// end = Rparen

	Synonym token.Pos // position of "SYNONYM" pseudo keyword
	Rparen  token.Pos // position of ")"

	Name    *Ident
```

**Review by makenowjust — COMMENTED** [review]
A small correction is needed.

File: ast/pos.go [hunk 2]
```diff
@@ -558,6 +558,9 @@ func (c *CreateTable) End() token.Pos {
 	return c.Rparen + 1
 }
 
+func (s *Synonym) Pos() token.Pos { return s.Synonym }
+func (s *Synonym) End() token.Pos { return s.Name.End() }
```
Comment (makenowjust) [inline ast/pos.go:562]:
```suggestion
func (s *Synonym) End() token.Pos { return s.Rparen + 1 }
```

**Comment by apstndb** [comment]
Thank you for your correction! All applied.

**Review by makenowjust — APPROVED** [review]
Thank you.

**Pull request merged**